Atomic/diff.py

```diff
@@ -43,8 +43,12 @@ def diff(self):
             # Clean up
             helpers._cleanup(image_list)
 
-            if self.args.json:
+            if self.args.json and not self.is_dbus:
                 util.output_json(helpers.json_out)
+
+            elif self.is_dbus:
+                return str(helpers.json_out)
+                
         except KeyboardInterrupt:
```

Comment:
This looks correct to me.  Basically he appears to be saying, if you want JSON and we arent using dbus, dump to stdout.  Thats what it originally did.